.github/workflows/publish-gh-pages.yml

```diff
@@ -0,0 +1,123 @@
+name: generate-docs-pages
+
+on:
+  push:
+    branches:
+      - main
```

Comment:
Do we want to do this on main or on a release branch? What if this fails for whatever reason?

Reply:
If it fails can run manually and gh_pages branch is unprotected worst case